### DISCUSSION TITLE
Connect with real thread devices

### DISCUSSION THREAD

**Discussion by Agatha2333** [openthread/openthread#8913] [Q&A]
Currently, I built an otbr on my raspberry pi with rpc dongle, and it runs well. I wonder how to add my eve thread device with my otbr. Could the otbr- agent add a "real" thread device?

Thank you for your consideration, and please let me know if you need any additional information from me.

**Comment by jwhui** [comment]
Getting Thread devices onto the same Thread network primarily involves configuring them with the same Thread Network Key.

Both iOS and Android provide APIs for sharing / retrieving Thread network credentials:
- https://developer.apple.com/documentation/threadnetwork
- https://developers.home.google.com/thread

Commissioning via Matter or HomeKit on these platforms make use of the shared Thread credentials.

**Reply by Agatha2333** [reply]
Thank you so much for your sharing. I still have questions about:
1)  Since we could build the OTBR on raspberry pi, is any tutorial about sharing/retrieving Thread network credentials via raspberry pi? Must I use IOS or Android provided APIs?
2) I also try to using chip-tool (Matter) to commission the eve thread devices, however, based on the link: https://github.com/project-chip/connectedhomeip/tree/master/examples/chip-tool , I could not get the discriminator and pairing code of the eve device.  How could I get the discriminator and pairing code of the real devices?

Thanks a lot for your reply!

**Reply by jwhui** [reply]
If you are not using the iOS / Android Matter APIs to pair the device, then you have some more flexibility.

There are various guides for pairing Matter devices using CHIPTool. You can find some in the [project-chip repo](https://github.com/project-chip/connectedhomeip/tree/master/docs/guides).

Regarding discriminator / pairing code, those should be available via the QR code.

Matter-specific questions are probably better raised within [https://github.com/project-chip/connectedhomeip](https://github.com/project-chip/connectedhomeip/).

**Reply by Agatha2333** [reply]
Thanks, but I am still a little bit confused about:
1) Could the thread device directly connect with CHIPTool? I think they must connect with the OTBR first, then they could be added by chip-tool. Based on the connection method, how could I pair the thread device(not support matter) with the OTBR(on a raspberry pi)? Please let me know if I have some misunderstanding about the connection.
2) I also made a picture of my eve thread device. The QR code is not similar to the example(virtual device).
![152341679949548_ pic](https://user-images.githubusercontent.com/62049491/228061112-8437c148-cae0-4057-ac6c-b731e30a818e.jpg)
![152351679949549_ pic](https://user-images.githubusercontent.com/62049491/228061267-fe7cd08d-0191-44bc-85f2-3ce9ca5674e6.jpg)

Thanks a lot.

**Reply by jwhui** [reply]
> Could the thread device directly connect with CHIPTool? I think they must connect with the OTBR first, then they could be added by chip-tool. Based on the connection method, how could I pair the thread device(not support matter) with the OTBR(on a raspberry pi)? Please let me know if I have some misunderstanding about the connection.

Did you review the documents I listed above? Here is one example that might give you a better idea: https://github.com/project-chip/connectedhomeip/blob/master/docs/guides/nrfconnect_android_commissioning.md#commissioning-nrf-connect-accessory-using-android-chiptool

> I also made a picture of my eve thread device. The QR code is not similar to the example(virtual device).

The picture shows a HomeKit QR code, which does not work with Matter. After upgrading your Eve device from HomeKit to Matter, you should have the option to download and save the Matter QR code.

**Reply by Agatha2333** [reply]
Thanks for responding! I carefully reviewed the information provided and I'm curious whether end devices in the thread network are required to use Matter or HomeKit in order to share their network key with the leader thread device, instead of simply pairing directly by pushing a button or scanning a code, like matter commissioning?

**Reply by jwhui** [reply]
Using Matter Commissioning requires the device to support Matter. Similarly, using HomeKit Commissioning requires the device to support HomeKit.
